### DISCUSSION TITLE
How stable is the archive format at this time (1/08/2022)?

### DISCUSSION THREAD

**Discussion by cliffordfajardo** [c9fe/22120#106] [Q&A] [answered]
Hi @i5ik,
Was curious to hear your perspective on the stability of the archiving format.
For example, if I'm using Diskernet today, and few months down the line the publishable search engine releases, do you think it will be compatible with the current archive format? Just curious 😄 

BTW, I know open source software development can often be hard & the fact that you published this is already a gift to the community

**Comment by crisdosaygo** [comment, marked as answer]
@cliffordfajardo thanks for this question, I can see you are thinking down the line. That's good.

I don't think I'll change the format, and I want it to be backward compatible (like Windows) if I do. 

The things that could change:

- hash function
- search engines

There might be others but that's the main ones.

Should not be a problem, because basically we will store some text that we use for indexing in the search engines on disk as well...so even if we changed the search engines, we could update an old archive to use the new search libraries by rebuilding the search indexes from the stored text.

The hash function change we could deal with by having a version, or some archive metadata that lists the hash function used in the archive.

Right now I have no plan to do this, but that could always change. 

And you've very cleverly and perceptively identified a possible risk in future--that's good!

Again, @cliffordfajardo thank you for your contributions to discussions! 🙂

**Comment by crisdosaygo** [comment]
Fairly stable. We just updated the hash used and probably need to ensure we read in expected sha and write out using the new one. So far we probably can't read existing archives with latest version. Will be fixed
